test/runner/job_test.exs

```diff
@@ -45,4 +19,253 @@ defmodule ElixirBench.Runner.JobTest do
     assert job.context.worker_num_cores in 1..64
     assert job.context.worker_os in [:Linux, :Windows, :macOS]
   end
+
+  describe "start_job/6" do
+    test "run job and return results given execute function" do
+      task_fun = fn job ->
+        {:ok, %{job | log: "success", status: 0}}
+      end
+
+      {:ok, job} = Job.start_job(job_fixture(), task_fun: task_fun)
+
+      assert %{log: "success", status: 0} = job
+    end
+
+    test "not start job if there is other job running" do
+      task_fun = fn _job ->
+        :timer.sleep(2_000)
+        {:ok, %{}}
+      end
+
+      %{pid: pid} =
+        Task.Supervisor.async_nolink(ElixirBench.Runner.JobsSupervisor, fn ->
+          :timer.sleep(2_000)
+        end)
+
+      assert_raise MatchError, fn ->
+        Job.start_job(job_fixture(), task_fun: task_fun)
+      end
+
+      Process.exit(pid, :kill)
+    end
+
+    test "handle job timeout" do
+      task_fun = fn _job ->
+        :timer.sleep(2_000)
+        {:ok, %{}}
+      end
+
+      job = Job.start_job(job_fixture(), task_fun: task_fun, timeout: 100)
+
+      assert %{log: "Job execution timed out", status: 127} = job
+    end
+  end
+
+  describe "ensure_no_other_jobs!/0" do
+    test "raise error if active child process already exist under JobsSupervisor" do
+      assert %{active: 0} = Job.ensure_no_other_jobs!()
+
+      %{pid: pid} =
+        Task.Supervisor.async_nolink(ElixirBench.Runner.JobsSupervisor, fn ->
+          :timer.sleep(3_000)
+        end)
+
+      assert_raise MatchError, fn ->
+        Job.ensure_no_other_jobs!()
+      end
+
+      Process.exit(pid, :kill)
+    end
+  end
+
+  describe "get_benchmars_output_path/1" do
```

Comment:
this should also probably be `benchmarks` :)

Reply:
fixed! :heavy_check_mark: